File: 2023-2024/domaci_kolo/templates/A.cs
// Upravte tuto funkci
int NejkratsiVzdalenost(int sirka, int delka, int predni, int zadni)
{
    int zleva = 0; // Vypočítejte cestu zleva
    int zprava = 0; // Vypočítejte cestu zprava

    return 0; // Určete kratší vzdálenost
}

// Načítání vstupu

void PrectiVstup(TextReader vstup, TextWriter vystup)
{
    int pocetProblemu = int.Parse(vstup.ReadLine()!);

    for (int i = 0; i < pocetProblemu; i++)
    {
        int sirka = int.Parse(vstup.ReadLine()!);
        int delka = int.Parse(vstup.ReadLine()!);
        int predni = int.Parse(vstup.ReadLine()!);
        int zadni = int.Parse(vstup.ReadLine()!);

        vystup.WriteLine(NejkratsiVzdalenost(sirka, delka, predni, zadni));
    }
}

(string, string)[] nazvySouboru = {
    ("A-lehky.txt", "A-lehky-vystup.txt"),
    ("A-tezky.txt", "A-tezky-vystup.txt"),
};

bool souborNalezen = false;

foreach (var (nazevVstupu, nazevVystupu) in nazvySouboru)
{
    try
    {
        using var vstup = new StreamReader(File.OpenRead(nazevVstupu));
        using var vystup = new StreamWriter(File.OpenWrite(nazevVystupu));

        PrectiVstup(vstup, vystup);

        souborNalezen = true;
    }
    catch (FileNotFoundException) { }
}

if (!souborNalezen)
{
    PrectiVstup(Console.In, Console.Out);
}

File: 2023-2024/domaci_kolo/templates/A.py
#!/usr/bin/env python3
import sys


def nejkratsi_vzdalenost(sirka, delka, predni, zadni):
    """Upravte tuto funkci"""

    zleva = 0  # Vypočítejte cestu zleva
    zprava = 0  # Vypočítejte cestu zprava

    return 0  # Určete kratší vzdálenost


# Načítání vstupu


def precti_vstup(input, output):
    pocet_problemu = int(input.readline())

    for _ in range(pocet_problemu):
        sirka = int(input.readline())
        delka = int(input.readline())
        predni = int(input.readline())
        zadni = int(input.readline())

        reseni = nejkratsi_vzdalenost(sirka, delka, predni, zadni)

        output.write(f"{reseni}\n")


NAZVY_SOUBORU = [
    ("A-lehky.txt", "A-lehky-vystup.txt"),
    ("A-tezky.txt", "A-tezky-vystup.txt"),
]

soubor_nalezen = False

for nazev_vstupu, nazev_vystupu in NAZVY_SOUBORU:
    try:
        with open(nazev_vstupu, "r", 1, "utf-8") as vstup:
            with open(nazev_vystupu, "w", 1, "utf-8") as vystup:
                precti_vstup(vstup, vystup)

        soubor_nalezen = True

    except FileNotFoundError:
        pass

if not soubor_nalezen:
    precti_vstup(sys.stdin, sys.stdout)
